src/live-programming.rst

```diff
@@ -36,8 +36,150 @@ Direct Manipulation Programming
+  Problems due to highly domain-specific knowledge requirements include the domain-specific knowledge about programming, mathematics, algorithms, and computer graphics.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'washburn2016went'
+
+  This paper presents an analysis of 155 postmortems published on the gaming site Gamasutra.com. 
+  The paper identifies characteristics of game development, links the characteristics to positive and negative experiences in the postmortems, and distills a set of best practices and pitfalls for game development.
+  Based on the analysis, the authors made the following recommendations to game developers: (1) Be sure to practice good risk management techniques;
+  (2) Prescribe to an iterative development process and utilize prototypes as a method of proving features and concepts before committing them to the design;
+  (3) Do not be overly ambitious in the design.
+
+As we can see in the above paper, there is a huge problem space in the game development domain including product (e.g. art, creativity, game design), development (e.g. tools, development process, team), resources (budge, hardware, publishers), and customer facing (e.g. marketing, feedback).
+This compendium covers only **tools** and **languages** for game development.
+
+Professional Platforms for Game Development
+-------------------------------------------------
+In recent years, game engines such as Unity3D and Unreal Engine have been the most popular platforms for game development.
+A game engine simplifies the task of the programmer by offering convenient abstractions for the hardware and operating systems on top of which the game runs.
+Game engines also provide reusable components such as physics, game inputs, rendering, and artificial intelligence (AI).
+Entity-component systems and event-handling mechanisms have been a fundamental feature of the software architecture of game engines nowadays.
+Entity-component systems allow users to attach components (i.e. scripts) to different game entities, which maximizes the reusability and interoperability of software modules.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'anderson2008case'
+
+  While available research has mainly focused on game engine subsystems such as rendering and AI, the issues regarding the overall architecture design of game engines have merely been discussed.
+  This article, in response, presents a number of key aspects and potential research topics with regard to the architecture design of game engines.
+  These topics include the establishment of a unified language of game development, the identification of software components that are common to all types of computer games, the definition of the role of content creation tools in the game development process and as part of game engines, and many others.
```

Comment:
"all types" -- is that what they claim in the paper? a more reasonable phrase would be "many types"